src/redis.nim

```diff
@@ -1014,7 +1014,10 @@ proc zrangebylex*(r: Redis | AsyncRedis, key: string, start: string, stop: strin
 proc zrank*(r: Redis | AsyncRedis, key: string, member: string): Future[RedisString] {.multisync.} =
   ## Determine the index of a member in a sorted set
   await r.sendCommand("ZRANK", key, @[member])
-  result = await r.readBulkString()
+  try:
+    result = $(await r.readInteger())
+  except ReplyError:
+    result = redisNil
```

Comment:
This looks error prone as a `ReplyError` can mean many different things, is there a function that you can use to read an integer or a null?

Reply:
There no function like that, I can update it to use Option or object variants

Reply:
I'll merge this, but if you could in a separate PR refactor this I would appreciate it.

Reply:
I will make another PR